Objects/typeobject.c

```diff
@@ -1925,6 +1925,8 @@ mro_invoke(PyTypeObject *type)
                                            &unbound);
         if (mro_meth == NULL)
             return NULL;
+        type->tp_flags &= ~(Py_TPFLAGS_HAVE_VERSION_TAG|
+                            Py_TPFLAGS_VALID_VERSION_TAG);
```

Comment:
This will disable caching for any class with a metaclass that is not `type`, even if it does not override `mro()`.
Changing `custom` to detect whether `mro()` has been overridden should fix that.

Reply:
Right after the `if (mro_meth == NULL) return;` ? I though it would returns if there's no mro_meth.

Reply:
Oh, you're right, it finds the mro method of the type object.